interfaces/builtin/thumbnailer.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const thumbnailerPermanentSlotAppArmor = `
+# Description: Allow use of aa_query_label API
+
+/sys/module/apparmor/parameters/enabled r,
+@{PROC}/@{pid}/mounts                   r,
+/sys/kernel/security/apparmor/.access   rw,
+`
+
+const thumbnailerConnectedSlotAppArmor = `
+# Description: Allow access to plug's data directory
+
+@{INSTALL_DIR}/###PLUG_SNAP_NAME###/**     r,
+owner @{HOME}/snap/###PLUG_SNAP_NAME###/** r,
+/var/snap/###PLUG_SNAP_NAME###/**          r,
+`
+
+type ThumbnailerInterface struct{}
+
+func (iface *ThumbnailerInterface) Name() string {
+	return "thumbnailer"
+}
+
+func (iface *ThumbnailerInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *ThumbnailerInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
```

Comment:
@jhenstridge, possibly, but I think it is too soon to tell. I mean, if you are acquiring a name, it is because you want to offer a service, so tying that to a connection for communicating does make sense. Let's see how people are using it in practice and see what needs refining.